tests/test_presence_do.py

```
@@ -0,0 +1,494 @@
+        stub.fetch.assert_awaited_once()
+
+    async def test_invalid_room_id_not_matched(self):
+        """Room IDs with special characters must NOT match the route."""
+        env = make_env()
+        env.CLASSROOM_DO = MagicMock()
+        env.PRESENCE_DO  = MagicMock()
+
+        req = MagicMock()
+        req.method = "GET"
+        req.headers = MagicMock()
+        req.headers.get = lambda k, d=None: None
```

Ruff (0.15.10)

[warning] Unused lambda argument: `k`

(ARG005)